src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -292,8 +314,16 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 			ereport(ERROR, (errmsg("grouped column list cannot be hashed")));
 		}
 
-		/* switch to hashed aggregate strategy to allow grouping */
-		aggregateStrategy = AGG_HASHED;
+		if (hasDistinctAggregate)
```

Comment:
Can we explain with a comment why do we need to use `AGG_SORTED` for distinct aggregate.